dist/dist_client.go

```diff
@@ -0,0 +1,500 @@
+		MinVersion:               tls.VersionTLS10,
+		PreferServerCipherSuites: true,
+		InsecureSkipVerify:       insecureSkipVerify,
+	}
+
+	certDir, err := ociCertDir(sys, hostName)
+	if err != nil {
+		return r, err
+	}
+	if err := tlsclientconfig.SetupCertificates(certDir, tlsClientConfig); err != nil {
+		return r, err
+	}
+
+	transport := &http.Transport{TLSClientConfig: tlsClientConfig}
+	client := &http.Client{Transport: transport}
+	creds := ""
+	if sys != nil {
+		if sys.DockerAuthConfig != nil {
+			a := sys.DockerAuthConfig
+			creds = base64.StdEncoding.EncodeToString([]byte(a.Username + ":" + a.Password))
+		} else {
+			registry := fmt.Sprintf("%s:%s", server, port)
+			if username, password, err := config.GetAuthentication(sys, registry); err == nil {
+				creds = base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+			}
+		}
+	}
+
+	r = OciRepo{ref: ref, authCreds: creds, client: client}
+
+	ping := func(scheme string) error {
+		u := url.URL{Scheme: scheme, Host: fmt.Sprintf("%s:%s", server, port), Path: fmt.Sprintf("/v2/")}
+		resp, err := client.Get(u.String())
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusUnauthorized {
+			return errors.Errorf("Error pinging registry %s:%s, response code %d (%s)",
+				server, port, resp.StatusCode, http.StatusText(resp.StatusCode))
+		}
+		return nil
+	}
+
+	scheme := "https"
+	err = ping(scheme)
+	if err != nil && insecureSkipVerify {
+		scheme = "http"
+		err = ping(scheme)
+	}
+	if err != nil {
+		return r, errors.Wrap(err, "unable to ping registry")
+	}
+
+	r.url = url.URL{Scheme: scheme, Host: fmt.Sprintf("%s:%s", server, port)}
+	return r, nil
+}
+
+func (o *OciRepo) GetManifest() ([]byte, *ispec.Manifest, error) {
+	name := o.ref.name
+	tag := o.ref.tag
+	m := &ispec.Manifest{}
+
+	var body []byte
+
+	manifestURI, err := url.Parse(fmt.Sprintf("/v2/%s/manifests/%s", name, tag))
+	if err != nil {
+		return body, m, errors.Wrapf(err, "couldn't parse manifest url for repo %s and image %s", name, tag)
+	}
+
+	manifestURI = o.url.ResolveReference(manifestURI)
+	req, err := http.NewRequest(http.MethodGet, manifestURI.String(), nil)
+
+	if err != nil {
+		return body, m, errors.Wrapf(err, "Couldn't create DELETE request for %v", manifestURI)
+	}
+
+	if o.authCreds != "" {
+		req.Header.Add("Authorization", "Basic "+o.authCreds)
+	}
+
+	req.Header.Add("Accept", "application/vnd.oci.image.manifest.v1+json")
+	resp, err := o.client.Do(req)
+
+	if err != nil {
+		return body, m, errors.Wrapf(err, "Error getting manifest %s %s from %v", name, tag, o.url)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return body, m, fmt.Errorf("bad return code %d getting manifest", resp.StatusCode)
+	}
+
+	body, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return body, m, errors.Wrapf(err, "Error reading response body for %s", tag)
+	}
+
+	err = json.Unmarshal(body, m)
```

Comment:
Why? Nothing uses the value AFAICS, and it’s, in general, not the job of the transport to worry about contents of the data (unless the transport is simulating something or doing internal conversions, I guess, like the `docker/tarfile` ones, and arguably that code should be moved into the generic conversion mechanism instead).